### PR TITLE
Drop support for PHP 7.0 and PHPUnit 6

### PR DESCRIPTION
**Pull request by oradwell**
As suggested by @theofidry, dropping support for PHP 7.0 and PHPUnit 6

Thanks to this, we can remove some compatibility code and use PHP 7.1 features

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 /vendor/
 /composer.lock
 .phpunit.result.cache
+.php-cs-fixer.cache
+php-cs-fixer.phar
```
Comment (theofidry) [inline .gitignore:5]: You can probably do `*.phar` to be honest

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
 /vendor/
 /composer.lock
 .phpunit.result.cache
+.php-cs-fixer.cache
```
Comment (theofidry) [inline .gitignore:4]: to keep with the convention above, I would recommend either removing the leading `/` or adding it here (this is a tiny nitpick feel free to ignore it)

File: .php-cs-fixer.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+$finder = PhpCsFixer\Finder::create()
+    ->in(__DIR__)
+;
+
+$config = new PhpCsFixer\Config();
```
Comment (theofidry) [inline .php-cs-fixer.php:9]: Not a big deal neither but in my experience introducing a CS fixer has always been quite noisy, might be worth extracting it to a different PR?

File: .travis.yml [hunk 4]
```diff
@@ -1,32 +1,33 @@
...
-  - 7.1
   - 7.2
   - 7.3
   - 7.4
   - 8.0
   - 8.1.0
 
 env:
-  - COVERAGE=0 INSTALL_PROPHECY=1
+  - COVERAGE=0 INSTALL_PROPHECY=1 CSCHECK=0
```
Comment (theofidry) [inline .travis.yml:11]:
```suggestion
  - COVERAGE=0 INSTALL_PROPHECY=1 CS_CHECK=0
```
?

File: composer.json [hunk 5]
```diff
@@ -16,9 +16,9 @@
         "psr-4": {"OckCyp\\CoversValidator\\Tests\\": "tests/"}
     },
     "require": {
-        "php": "^7.0 || ^8.0",
-        "phpunit/phpunit": "^6.0 || ^7.0 || ^8.0 || ^9.0",
-        "symfony/console": "^2.7 || ^3.0 || ^4.0 || ^5.0 || ^6.0"
+        "php": ">=7.1",
```
Comment (theofidry) [inline composer.json:19]: note that you're also making it compatible with PHP9 this way

Comment (theofidry) [inline composer.json:19]: is there a reason to not drop even lower the versions? 7.4 is going EOL soon so you could put it as minimum and still be considered very lenient in terms of support

**Review by theofidry — APPROVED** [review]

File: composer.json (hunk 5 above)
Comment (oradwell) [reply composer.json:19]: I couldn't find any compatibility code necessary for PHP versions between 7.1 and 7.4. PHP 8.0 will bring the "mixed" type, so I guess we can drop more versions when PHP 8.0 is as popular as PHP 7.4 is now

Comment (oradwell) [reply composer.json:19]: some libraries seem to do it this way. I'll revert
```suggestion
        "php": "^7.0 || ^8.0",
```

File: .php-cs-fixer.php (hunk 3 above)
Comment (oradwell) [reply .php-cs-fixer.php:9]: it's still small enough, so I'll leave it. I'll keep it in mind for the next time. I only added it, so we actually start using some PHP 7.1 features
